compact.go

```diff
@@ -476,14 +486,21 @@ func (w *instrumentedChunkWriter) WriteChunks(chunks ...chunks.Meta) error {
 func (c *LeveledCompactor) write(dest string, meta *BlockMeta, blocks ...BlockReader) (err error) {
 	dir := filepath.Join(dest, meta.ULID.String())
 	tmp := dir + ".tmp"
-
+	var closers []io.Closer
 	defer func(t time.Time) {
+		var merr MultiError
+		merr.Add(err)
+		err = merr.Err()
```

Comment:
This should come after the `for` loop below. Or rather, we can reuse `closeAll` here.

Reply:
updated, thanks!